Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java

```diff
@@ -225,25 +227,39 @@ public void showInterstitialAd(final MaxAdapterResponseParameters parameters, fi
 
     public void loadAppOpenAd(final MaxAdapterResponseParameters parameters, @Nullable final Activity activity, final MaxAppOpenAdapterListener listener)
     {
+        String appId = parameters.getServerParameters().getString( "app_id", null );
+        if ( TextUtils.isEmpty( appId ) )
+        {
+            log( "No app ID provided" );
+            listener.onAppOpenAdLoadFailed( MaxAdapterError.INVALID_CONFIGURATION );
+            return;
+        }
+
         String bidResponse = parameters.getBidResponse();
         boolean isBiddingAd = AppLovinSdkUtils.isValidString( bidResponse );
         String placementId = parameters.getThirdPartyAdPlacementId();
+        Context context = getContext( activity );
         log( "Loading " + ( isBiddingAd ? "bidding " : "" ) + "app open ad for placement: " + placementId + "..." );
 
-        if ( !VungleAds.isInitialized() )
-        {
-            log( "Vungle SDK not successfully initialized: failing app open ad load..." );
-            listener.onAppOpenAdLoadFailed( MaxAdapterError.NOT_INITIALIZED );
-
-            return;
-        }
-
         updateUserPrivacySettings( parameters );
 
-        appOpenAd = new InterstitialAd( getContext( activity ), placementId, new AdConfig() );
-        appOpenAd.setAdListener( new AppOpenAdListener( listener ) );
+        VungleInitializer.getInstance().initialize( appId, context, new VungleInitializationListener()
+        {
+            @Override
+            public void onInitializeSuccess()
+            {
+                appOpenAd = new InterstitialAd( context, placementId, new AdConfig() );
```

Comment:
QQ What happens if the sdk is already initialized. will this be triggered.

Reply:
Yes, the onInitializeSuccess () will be triggered immediately.
https://github.com/Vungle/AppLovin-MAX-SDK-Android/pull/20/files#diff-0592542557c7e0c6b02d7f79bd19b302b1ae69441b5cd26dc567c138ca1a1049R33-R35